test/integration/sequelize/deferrable.test.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+/* jshint -W030 */
+/* jshint -W110 */
+var _ = require('lodash')
+  , chai = require('chai')
+  , expect = chai.expect
+  , assert = chai.assert
+  , Support = require(__dirname + '/../support')
+  , dialect = Support.getTestDialect()
+  , Sequelize = require(__dirname + '/../../../index')
+  , config = require(__dirname + '/../../config/config')
+  ;
+
+describe(Support.getTestDialectTeaser('Sequelize'), function() {
+  if (dialect !== 'postgres') {
+    return;
+  }
+
+  beforeEach(function () {
+    this.run = function (deferrable, options) {
+      options = options || {};
+
+      var taskTableName      = options.taskTableName || 'tasks_' + config.rand();
+      var transactionOptions = _.extend({ deferrable: Sequelize.Deferrable.SET_DEFERRED }, options);
+      var userTableName      = 'users_' + config.rand();
+
+      var User = this.sequelize.define(
+        'User', { name: Sequelize.STRING }, { tableName: userTableName }
+      );
+
+      var Task = this.sequelize.define(
+        'Task', {
+          title: Sequelize.STRING,
+          user_id: {
+            allowNull: false,
+            type: Sequelize.INTEGER,
+            references: userTableName,
+            referencesKey: 'id',
+            referencesDeferrable: deferrable
+          }
+        }, {
+          tableName: taskTableName
+        }
+      );
+
+      return User.sync({ force: true }).bind(this).then(function () {
+        return Task.sync({ force: true });
+      }).then(function () {
+        return this.sequelize.transaction(
+          transactionOptions,
+          function (t) {
+            return Task
+              .create({ title: 'a task', user_id: -1 }, { transaction: t })
+              .then(function (task) {
+                return [task, User.create({}, { transaction: t })];
+              })
+              .spread(function (task, user) {
+                task.user_id = user.id;
+                return task.save({ transaction: t });
+              });
```

Comment:
I fixed for the first idea. Didn't fix for the second though. IMHO it's super nicely readable if the code flow call are indented the way it is. 

So instead of 

```
return Task.find().then(function () { 
  return whatever
})
```

I prefer

```
return Task
  .find()
  .then()
```

@mickhansen @janmeier ?